pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/StarTreeIndexOperator.java

```diff
@@ -360,7 +377,8 @@ public BlockId getId() {
       HashSet<String> remainingPredicateColumns = searchEntry.remainingPredicateColumns;
       HashSet<String> remainingGroupByColumns = searchEntry.remainingGroupByColumns;
       // Check if its leaf
```

Comment:
update the comment, its terminating if its leaf or (all filters are applied and aggregatedDocId is valid)